src/openrct2/object/Object.cpp

```diff
@@ -329,6 +329,55 @@ std::unique_ptr<IStream> ObjectAsset::GetStream() const
     return {};
 }
 
+u8string VersionString(const ObjectVersion& version)
+{
+    return std::to_string(std::get<0>(version)) + "." + std::to_string(std::get<1>(version)) + "."
+        + std::to_string(std::get<2>(version));
+}
+
+ObjectVersion VersionTuple(std::string_view version)
+{
+    if (version.empty())
+    {
+        return std::make_tuple(0, 0, 0);
+    }
+
+    auto nums = String::Split(version, ".");
+    uint16_t versions[VersionNumFields] = {};
+    if (nums.size() > VersionNumFields)
+    {
+        log_warning("%i fields found in version string '%s', expected X.Y.Z", nums.size(), version);
+    }
+    if (nums.size() == 0)
+    {
+        log_warning("No fields found in version string '%s', expected X.Y.Z", version);
+        return std::make_tuple(0, 0, 0);
+    }
+    // fill in array from right-to-left
+    try
+    {
+        uint8_t highestIndex = std::min(static_cast<int8_t>(nums.size()), VersionNumFields);
```

Comment:
This is quite a mixup between types. Please make `highestIndex`, `i` and `VersionNumFields` size_t. That should also allow you to get rid of the cast.

Reply:
I seem to remember some issue with size_t and arrays, but it could have been issues with the segfaults. I will try it again.